Objective-C/Internal/Replicator/CBLTrustCheck.mm

```diff
@@ -173,8 +173,19 @@ - (NSURLCredential*) checkTrust: (NSError**)outError {
 
     // If using cert-pinning, accept cert iff it matches the pin:
     if (_pinnedCertData) {
-        SecCertificateRef cert = SecTrustGetCertificateAtIndex(_trust, 0);
-        if ([_pinnedCertData isEqual: CFBridgingRelease(SecCertificateCopyData(cert))]) {
+        SecCertificateRef cert = NULL;
+#if __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000 || __IPHONE_OS_VERSION_MIN_REQUIRED >= 150000
+        if (@available(macOS 12.0, iOS 15.0, *)) {
```

Comment:
Do we still need `if (@available(macOS 12.0, iOS 15.0, *)) {`?

Reply:
When I looked at this again, the fix will have side effect because we will left out anything in #else completely when we built the binary. #if __IPHONE_OS_VERSION_MIN_REQUIRED is for compile time checking of the min target SDK version.